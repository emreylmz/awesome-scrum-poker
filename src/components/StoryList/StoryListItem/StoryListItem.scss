@import "../../../styles/colors";
@import "../../../styles/metrics";

.story-list-item {
  display: flex;
  flex-direction: row;
  flex: 1;
  align-content: space-around;

  padding: $default-margin;

  &__name {
    display: flex;
    flex: 1;
  }

  &__point {
    display: flex;
    flex: 1;

  }

  &__status {
    display: flex;
    flex: 1;

  }
}
