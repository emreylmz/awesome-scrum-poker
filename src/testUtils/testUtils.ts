export const emptyTestFunc = (): void => void 0
