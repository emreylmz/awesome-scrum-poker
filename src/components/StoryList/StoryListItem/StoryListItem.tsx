import React from 'react'
import { Render } from '../../../types'
import { classNames } from '../../../utils'
import './StoryListItem.scss'

interface Props {
  title: string
  point?: number
  status: number
  isOdd?: boolean
}

class StoryListItem extends React.PureComponent<Props> {
  getStatusName(status: number): string {
    switch (status) {
      case 0:
        return 'Not Voted'
      case 1:
        return 'Active'
      case 2:
        return 'Voted'
      default:
        return ''
    }
  }
  render(): Render {
    const { title, point, status, isOdd } = this.props
    const className = classNames('story-list-item', isOdd && 'light-background')
    return (
      <div className={className}>
        <div className={'story-list-item__name'}>{title}</div>
        <div className={'story-list-item__point'}>{point}</div>
        <div className={'story-list-item__status'}>{this.getStatusName(status)}</div>
      </div>
    )
  }
}

export type StoryListItemProps = Props
export default StoryListItem
