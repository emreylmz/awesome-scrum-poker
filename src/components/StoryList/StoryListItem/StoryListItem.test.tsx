import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import '../../../testUtils/setupTests'
import StoryListItem, { StoryListItemProps } from './StoryListItem'

type DoneCallback = jest.DoneCallback

const defaultProps: StoryListItemProps = {
  title: 'Default Story',
  status: 0
}

const getWrapper = (props?: StoryListItemProps): ShallowWrapper => {
  return shallow(<StoryListItem {...defaultProps} {...props} />)
}

describe('Story List Item UI', (): void => {
  it('should render correctly', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it('should change story name with props', (done: DoneCallback): void => {
    done()
  })

  it('should change story active status with props', (done: DoneCallback): void => {
    done()
  })

  it('should change story point with props', (done: DoneCallback): void => {
    done()
  })
})
