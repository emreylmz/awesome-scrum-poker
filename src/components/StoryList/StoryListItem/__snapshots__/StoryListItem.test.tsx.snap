// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story List Item UI should render correctly 1`] = `ShallowWrapper {}`;
